extension-base/src/main/java/com/azure/autorest/extension/base/plugin/CadlSettings.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.autorest.extension.base.plugin;
+
+public class CadlSettings {
```

Comment:
why do we have CadlSetting in shared module? Should it be in cadl-extension module?

Reply:
I'm thinking about a global way of getting cadl settings like `output-folder`. Previously we had `JavaSettings.getAutorestSettings().getOutputFolder()`. I intended to add `JavaSettings.getCadlSettings().getOutputFolder()`. 

For now, the only place it is used is in `CadlPlugin`, so it seems too much to add a new class for it. Will inline the property in `CadlPlugin` instead.